lib/content-services/tree-view/models/tree-view.model.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * @license
+ * Copyright 2016 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NodeEntry } from 'alfresco-js-api';
+
+export class TreeBaseNode {
+
+    name: string;
+    nodeId: string;
+    level: number;
+    expandable = true;
+
+    constructor(obj: NodeEntry, level?: number) {
```

Comment:
can we call it nodeEntry instead of obj ? 

plus if we store also the full nodeentry in the model after we can emit from the component. Emitting the nodeEntry will help the interoperability of the component. The nodeEntry structure is quite common in many of our input it will be a good gain